Rearrange Docker docs

davidcassany opened the pull request:
This commit rearranges the Docker image builds documentation
in order to match the criteria exposed in #323.

schaefi commented:
Thanks David

@tomschr could you have a short look ?

tomschr commented on doc/source/building/build_docker_container.rst:
```diff
@@ -1,190 +1,91 @@
-can be specified:
+The Docker configuration metadata is provided to KIWI as part of the
+:ref:`XML description file <decription_components>` using the
+``<containerconfig>`` section. The following parameters
```
As we talk about XML, we should better use the term "tag" or "element" instead of "section".

tomschr replied:
Same with "parameters". I guess, these are (XML) attributes, right? If yes, better use "attributes" here.

tomschr commented on doc/source/building/build_docker_container.rst:
```diff
@@ -1,190 +1,91 @@
 * ``expose``: Informs at which ports is the container listening at runtime.
-* ``env``: Sets an environment value, is key-value pair.
+  Equivalent of the `EXPOSE` directive of a Docker file.
+* ``environment``: Sets an environment values using key-value pairs.
```
Plural or singular? I assume plural here, in that case I would rephrase to:

    Set environment values using key-value pairs.

It would be helpful to add *how* a single key-value pair can be separated from another one. Use spaces, commas, or something different?

tomschr commented on doc/source/building/build_docker_container.rst:
```diff
@@ -1,190 +1,91 @@
-  `entrypoint`.
+  include parameters. Equivalent of the `ENTRYPOINT` directive of a Docker
+  file..
+* ``volumes``: Creates mountpoints with the given name and marks it to hold
```
Create mountpoints...

tomschr commented on doc/source/building/build_docker_container.rst:
```diff
@@ -1,190 +1,91 @@
+* ``volumes``: Creates mountpoints with the given name and marks it to hold
+  external volumes from the host or from other contianers. Equivalent of the
+  `VOLUME` directive of a Docker file.
+* ``user``: Sets the user name or UID to be used when running `entrypoint` and
```
I would slightly rephrase it as:  "Sets the user name or user id (UID)..."

tomschr commented on doc/source/building/build_docker_container.rst:
```diff
@@ -1,190 +1,91 @@
-  or by including an `<archive>` item inside the `<packages type="image">`
-  section. **ADD** also handles remote content, in KIWI this should be done
-  within the `config.sh` script.
+Other Docker file directives such as ``RUN`` and ``COPY`` or ``ADD``, can be
```
Correct with:

  ... ``RUN``, ``COPY`` or ``ADD``, ...

tomschr commented on doc/source/building/build_docker_container.rst:
```diff
@@ -1,190 +1,91 @@
+   see :ref:`example-descriptions`.
 
-* `COPY`
+2. Install the required toolchain for Docker images building.
```
IMHO, I would separate the step of including the repo from installing the packages like this:

```
1. Include the ``Virtualization`` repository to your list:

   .. code:: bash
      $ zypper ar -f http://download.opensuse.org/repositories/Virtualization:/containers/<DIST>

2. Install the :command:`umoci` and :command:`skopeo` tools:
  [...]
```

tomschr commented on doc/source/building/build_docker_container.rst:
```diff
@@ -1,190 +1,91 @@
-``derived_from="docker://opensuse:leap"``). The following example
-type specification shows how to specify a base image setup:
+   where the placeholder `<DIST>` is the preferred distribution. Then install
+   :command:`umoci` and :command:`skopeo` tools
```
See [previous comment](https://github.com/SUSE/kiwi/pull/363/files#r120322349).

davidcassany commented:
@tomschr I think I have addressed all your comments, I have divided the list of configurable parameters between attributes and tags, I think this way is more clear. I did not want to document how to explicitly configure the Docker image, I assume that the users of this feature are already familiar with the Dockerfile syntax, so I just wanted to give an overview at a glance of the configuration possibilities, rather than explain them in detail.

schaefi commented:
recompiled the docs and to me it looks great 👍

tomschr commented:
@davidcassany 
>  I think I have addressed all your comments,

Great, thank you for your efforts!

>  I have divided the list of configurable parameters between attributes and tags, I think this way is more clear.

Yes, much better! 👍

> I did not want to document how to explicitly configure the Docker image, I assume that the users of this feature are already familiar with the Dockerfile syntax, so I just wanted to give an overview at a glance of the configuration possibilities, rather than explain them in detail.

Yes, totally agree. If you like and know a good link (maybe upstream Docker docu?) you can reference it and delegate it to to remote documentation. No need to copy existing texts.

tomschr commented:
@davidcassany, @schaefi after I've built the doc myself, I have a final question. 😉

How should we display commands that require root privileges?

In most cases, we use `$` as a prompt. IMHO it could be misleading. Although the experienced user would probably know how to deal with it, nonetheless we should be clear about that. I see the following solutions to it:

1. Use `#` to indicate the root prompt.
2. Use `sudo`.
3. Leave it as is and assume all users are experienced

This is mostly important for the `zypper` and `docker` commands, depending on how the user configured its system. Regardless what we use, we should be consistent. 😉

What do you think?

schaefi commented:
I prefix any command with a '$' and for those which require sudo i write '$ sudo'. I'm not sure if this follows the common practice but I did that all over the documentation

tomschr commented:
> I prefix any command with a '$' and for those which require sudo i write '$ sudo'.
> I'm not sure if this follows the common practice but I did that all over the documentation.

The only "common practice" is to be consistent. 😉 It doesn't matter if it is with `sudo` or with `#` as long as the reader knows what to do and what to expect.

So no need to change anything here. I just stumbled upon it. Sorry for the noise.
